docs/adrs/006-web-apis.md

```diff
@@ -0,0 +1,240 @@
+- Estimate transaction costs
+- Stake and swap assets
+- Participate in protocol governance
+
+This list is supposed to be covered by the proposed API, documented and presented with examples to users.
+
+## Creating the client
+
+In other blockchains, JavaScript SDKs usually split the interfaces into at least two parts: one that manages the wallet connection and private transactions, and one that reads the public blockchain data. Examples include [Viem](https://viem.sh/docs/clients/intro) and [Thirdweb](https://portal.thirdweb.com/typescript/v5/client) on Ethereum.
+
+In Penumbra, any blockchain query requires a connection with the wallet provider, therefore all actions and
+queries can be called 'private'. It is reasonable to construct the notion of a **client** – the interface 
+that manages the injected connections and provides methods for interacting with the blockchain.
+
+Creating the `client` should be the starting point for any application working with Penumbra:
+
+```ts
+import { createPenumbraClient } from '@penumbra-zone/client';
+
+export const client = createPenumbraClient();
+```
+
+The flow of work with the `client` would be as follows:
+
+- User creates an instance of the `client`
+- They use the `client` to establish the injected connection, and the `client` saves the connection in its state
+- To fetch information from the wallet, the application passes the `client` instance to the fetch methods
+
+The idea of a single shared `client` is inspired by these popular TypeScript libraries:
+- [Prisma ORM](https://www.prisma.io/docs/orm/prisma-client/setup-and-configuration/instantiate-prisma-client)
+- [Supabase](https://supabase.com/docs/reference/javascript/initializing)
+- [Viem](https://viem.sh/docs/clients/intro)
+
+## Injected connection
+
+The `client` must be able to connect to injected wallets. To make the API possible, the interface injected into the `window` object must be standardized:
+
+```ts
+export interface PenumbraProvider extends Readonly<PenumbraStateEventTarget> {
+  /** Should contain a URI at the provider's origin, 
+   * serving a manifest describing this provider */
+  readonly manifest: string;
+
+  /** Call to gain approval. Returns `MessagePort` to this provider. 
+   * Might throw descriptive errors if user denies the connection */
+  readonly connect: () => Promise<MessagePort>;
+
+  /** A synchronous call to retrive the `MessagePort` if the connection was already established */
+  readonly reconnect: () => MessagePort | undefined;
+
+  /** Call to indicate the provider should discard approval of this origin. 
+   * Successfull if doesn't throw errors */
+  readonly disconnect: () => Promise<void>;
+
+  /** Should synchronously return the present connection state.
+   * - `true` indicates active connection.
+   * - `false` indicates connection is closed or rejected.
+   * - `undefined` no attempt has resolved. Connection may be attempted
+   */
+  readonly isConnected: () => boolean | undefined;
```

Comment:
Changed to `boolean`. The `ConnectedState` can be accessed via `state` function as a more descriptive one

Reply:
i want to keep this aspect of the api.

it is intuitive that 'truthy value means yes, falsy value means no'.

the additional information of 'maybe' and 'never' is available by inspection of the falsy value.

- 'undefined' is used to represent a state that is not yet resolved - it should become 'true' or 'false' later
- 'false' is used to represent a state that is resolved - it will definitely remain 'false'

i think this is simple to explain and simple to understand. i find it very useful to have a simply truthy/falsy check.

detailed state is available from the state method.

Reply:
Sorry, this is not a good API design. We should not assign meaning to `undefined` and cause our consumers to try to guess what it means and hope they infer like we have. We should replace this with an enum if there are more possible states than two.

Reply:
> 'false' is used to represent a connection that is rejected - it will definitely remain 'false'

@turbocrime why can't users re-establish the connection when `isConnected = false`?